pkg/yurttunnel/handlerwrapper/tracerequest/tracereq.go

```diff
@@ -17,29 +17,58 @@ limitations under the License.
 package tracerequest
 
 import (
+	"errors"
+	"net"
 	"net/http"
 	"time"
 
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
 	hw "github.com/openyurtio/openyurt/pkg/yurttunnel/handlerwrapper"
 	"github.com/openyurtio/openyurt/pkg/yurttunnel/server/metrics"
 )
 
+const (
+	proxyHostHeaderKey = "X-Tunnel-Proxy-Host"
+)
+
 // TraceReqMiddleware prints request information when start/stop
 // handling the request
-type traceReqMiddleware struct{}
+type traceReqMiddleware struct {
+	nodeLister      corelisters.NodeLister
+	informersSynced []cache.InformerSynced
+}
 
 // NewTraceReqMiddleware returns an middleware object
 func NewTraceReqMiddleware() hw.Middleware {
-	return &traceReqMiddleware{}
+	return &traceReqMiddleware{
+		informersSynced: make([]cache.InformerSynced, 0),
+	}
 }
 
 func (trm *traceReqMiddleware) Name() string {
 	return "TraceReqMiddleware"
 }
 
+// SetSharedInformerFactory set nodeLister and nodeSynced for WrapHandler
+func (trm *traceReqMiddleware) SetSharedInformerFactory(factory informers.SharedInformerFactory) error {
```

Comment:
Where is the caller of this method?

Reply:
It is called by `pkg/yurttunnel/handlerwrapper/initializer/initalizer.go#Initialize()`